test/unit/runtime-rules-hotloading.test.ts

```diff
@@ -0,0 +1,96 @@
+import * as path from 'path';
+import * as loadFilterRules from '../../lib/filter-rules-loading';
+
+const scmRulesToTest = [
+  'azure-repos',
+  'bitbucket-server',
+  'github-com',
+  'github-enterprise',
+  'gitlab',
+];
+
+describe('filter Rules Loading', () => {
+  test.each(scmRulesToTest)(
+    'Loads normal accept file - Testing %s',
+    (folder) => {
+      const loadedRules = loadFilterRules(
+        'accept.json.sample',
+        path.join(__dirname, '../..', `client-templates/${folder}`),
+      );
+
+      expect(loadedRules).toMatchSnapshot();
+    },
+  );
+
+  test.each(scmRulesToTest)(
+    'Skip injection if no or invalid IAC extensions - Testing %s',
+    (folder) => {
+      process.env.ACCEPT_IAC = 'rf';
+      const loadedRules = loadFilterRules(
+        'accept.json.sample',
+        path.join(__dirname, '../..', `client-templates/${folder}`),
+      );
+
+      expect(loadedRules).toMatchSnapshot();
+      delete process.env.ACCEPT_IAC;
+    },
+  );
+
+  test.each(scmRulesToTest)(
+    'Injection of valid IAC extensions - Testing %s',
+    (folder) => {
+      process.env.ACCEPT_IAC = 'tf,yaml, json,yml,tpl';
+      const loadedRules = loadFilterRules(
+        'accept.json.sample',
+        path.join(__dirname, '../..', `client-templates/${folder}`),
+      );
+
+      expect(loadedRules).toMatchSnapshot();
+      delete process.env.ACCEPT_IAC;
+    },
+  );
+
+  test.each(scmRulesToTest)(
+    'Injection of valid IAC extensions - Testing %s',
+    (folder) => {
+      process.env.ACCEPT_IAC = 'tf,json';
+      const loadedRules = loadFilterRules(
+        'accept.json.sample',
+        path.join(__dirname, '../..', `client-templates/${folder}`),
+      );
+
+      expect(loadedRules).toMatchSnapshot();
+      delete process.env.ACCEPT_IAC;
+    },
+  );
+
+  test.each(scmRulesToTest)(
+    'Injection of valid CODE rules - Testing %s',
+    (folder) => {
+      process.env.ACCEPT_CODE = 'true';
+      const loadedRules = loadFilterRules(
+        'accept.json.sample',
+        path.join(__dirname, '../..', `client-templates/${folder}`),
+      );
+
+      expect(loadedRules).toMatchSnapshot();
```

Comment:
small thing: since you only really care about `code` rules here, you could explicitly filter those two json entries from `loadedRules` - by, for example, filtering for objects that have `for git clone` string and only then snapshot match. That way, you will have less bloated snapshot file.

Reply:
Doing the whole thing makes me feel better than there is no side effects on the rest of the accept files. Since copy vs nestedcopy gets a bit tricky, I'd rather being comprehensive even at the cost of bloating snapshots

Reply:
makes sense